components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -5553,57 +5566,163 @@ public void deployAPIRevision(String apiId, String apiRevisionId,
-                .addAndRemovePublishedGatewayLabels(apiId, apiRevisionId, environmentsToAdd, gatewayVhosts,
-                        environmentsToRemove);
-        apiMgtDAO.addAPIRevisionDeployment(apiRevisionId, apiRevisionDeployments);
-        if (environmentsToAdd.size() > 0) {
-            // TODO remove this to organization once the microgateway can build gateway based on organization.
-            gatewayManager.deployToGateway(api, organization, environmentsToAdd);
+
+        apiMgtDAO.addAPIRevisionDeployment(apiRevisionUUID, apiRevisionDeployments);
+
+        WorkflowExecutor revisionDeploymentWFExecutor = getWorkflowExecutor(
+                WorkflowConstants.WF_TYPE_AM_REVISION_DEPLOYMENT);
+
+        for (APIRevisionDeployment apiRevisionDeployment : apiRevisionDeployments) {
+            APIRevisionWorkflowDTO revisionWFDto = new APIRevisionWorkflowDTO();
+            try {
+                revisionWFDto.setAPIRevision(apiRevision);
+                revisionWFDto.setExternalWorkflowReference(revisionDeploymentWFExecutor.generateUUID());
+                revisionWFDto.setWorkflowReference(apiRevisionUUID);
+                revisionWFDto.setWorkflowType(WorkflowConstants.WF_TYPE_AM_REVISION_DEPLOYMENT);
+                revisionWFDto.setCallbackUrl(revisionDeploymentWFExecutor.getCallbackURL());
+                revisionWFDto.setStatus(WorkflowStatus.CREATED);
+                revisionWFDto.setTenantDomain(organization);
+                revisionWFDto.setTenantId(tenantId);
+                revisionWFDto.setCreatedTime(System.currentTimeMillis());
+                revisionWFDto.setUserName(userNameWithoutChange);
+                revisionWFDto.setApiName(apiIdentifier.getApiName());
+                revisionWFDto.setApiVersion(apiIdentifier.getVersion());
+                revisionWFDto.setApiProvider(apiIdentifier.getProviderName());
+                revisionWFDto.setEnvironment(apiRevisionDeployment.getDeployment());
+                revisionWFDto.setRevisionId(String.valueOf(revisionId));
+                revisionDeploymentWFExecutor.execute(revisionWFDto);
+            } catch (WorkflowException e) {
+                log.error("Unable to execute Revision Deployment Workflow for revision: " + apiRevisionUUID, e);
+                handleException("Unable to execute Revision Deployment Workflow for revision: " + apiRevisionUUID, e);
+            }
+
+            // get the workflow state once the executor is executed.
+            WorkflowDTO wfDTO = apiMgtDAO.retrieveWorkflow(revisionWFDto.getExternalWorkflowReference());
+            WorkflowStatus apiWFState = null;
+            if (wfDTO != null) {
+                apiWFState = wfDTO.getStatus();
+            }
+            // if the workflow has started, then executor should not fire again
+            if (WorkflowStatus.CREATED.equals(apiWFState)) {
+                try {
+                    revisionDeploymentWFExecutor.complete(wfDTO);
```

Comment:
Why is this still needed? complete() is already called from APIRevisionDeploymentSimpleWorkflowExecutor https://github.com/wso2/carbon-apimgt/pull/12169/files#diff-fb6b4b3f12e41a84af4ea0a7f867f3656862c5c730212e2fd60e0f4ba493ae3eR60

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12169/commits/932d40dfc0938ed2b1e93e152eb930615b9a04e5